zipline/utils/events.py

```diff
@@ -495,18 +507,16 @@ def get_first_trading_day_of_week(dt, env):
         # trading day of the env is also the first trading day of the
         # week(in the TradingEnvironment, at least), so just return
         # that date.
-        if dt is None:
-            return prev
-        while dt.date().weekday() < prev.date().weekday():
+        while dt and dt.date().weekday() < prev.date().weekday():
             prev = dt
             dt = env.previous_trading_day(dt)
-            if dt is None:
-                return prev
 
         if env.is_trading_day(prev):
             return prev.date()
-        else:
+        elif env.next_trading_day(prev):
             return env.next_trading_day(prev).date()
```

Comment:
How fast is `next_trading_day`?  Should we cache the result instead of calling it twice here?